src/Microsoft.AspNetCore.Razor.Language/DefaultRazorParserOptions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    internal class DefaultRazorParserOptions : RazorParserOptions
+    {
+        public DefaultRazorParserOptions(DirectiveDescriptor[] directives, bool designTime, bool parseOnlyLeadingDirectives)
+        {
+            if (directives == null)
+            {
+                throw new ArgumentNullException(nameof(directives));
+            }
+
+            Directives = directives;
+            DesignTime = designTime;
+            ParseOnlyLeadingDirectives = parseOnlyLeadingDirectives;
+        }
+
+        public override bool DesignTime { get; }
+
+        public override IReadOnlyCollection<DirectiveDescriptor> Directives { get; }
+
+        public override bool ParseOnlyLeadingDirectives { get; }
```

Comment:
I renamed this option because it's really intended to parse ALL leading directives, not just the first line. I'll love an issue